### PR TITLE
Search through all shards when searching by hash

### PR DESCRIPTION
**Pull request by priyawadhwa**
This adds in a test with `cosign sign-blob` and `cosign verify-blob` to make sure it is working as expected

### CODE REVIEW

**Comment by Hayden-IO** [comment]
Do we need to block on the Cosign fix? It's fairly unlikely that someone has uploaded the same entry cross-shard - It likely only affects the individual who reported the issue, and we can find a workaround (re-sign and reupload).

File: pkg/api/entries.go [hunk 1]
```diff
@@ -415,23 +415,27 @@ func SearchLogQueryHandler(params entries.SearchLogQueryParams) middleware.Respo
...
+					leafResult := resp.getLeafAndProofResult
+					if leafResult != nil && leafResult.Leaf != nil {
+						results = append(results, resp.getLeafAndProofResult)
+					}
 				}
-				leafResult := resp.getLeafAndProofResult
-				if leafResult != nil && leafResult.Leaf != nil {
-					searchByHashResults[i] = leafResult
+				if results == nil {
+					return fmt.Errorf("no responses found")
```
Comment (Hayden-IO) [inline pkg/api/entries.go:442]: do we need to set response code to `http.StatusNotFound`?

**Comment by priyawadhwa** [comment]
@haydentherapper yep! removed that comment, this should be good to merge once it's green

**Review by dlorenc — DISMISSED** [review]

File: tests/sharding-e2e-test.sh [hunk 2]
```diff
@@ -24,6 +24,8 @@ set -ex
 echo "Installing createtree..."
 go install github.com/google/trillian/cmd/createtree@latest
 
+echo "Installing cosign..."
+go install github.com/sigstore/cosign/cmd/cosign@latest
```
Comment (bobcallaway) [inline tests/sharding-e2e-test.sh]: this logic may evolve as time goes on - are we better actually reimplementing the logic here by directly calling the rekor APIs?

Comment (priyawadhwa) [reply tests/sharding-e2e-test.sh]: i think we probably want to test with cosign somewhere, whether it's here or in a prober. not super opinionated on this one.

**Review by asraa — COMMENTED** [review]
server changes look fine to me, but CLI logic is incorrect now, for e.g.
https://github.com/sigstore/rekor/blob/204d9afd1ba326284f9c4a91d92358ade513120e/cmd/rekor-cli/app/verify.go#L140

just create a follow-up issue, because this is imp

File: tests/sharding-e2e-test.sh (hunk 2 above)
Comment (asraa) [reply tests/sharding-e2e-test.sh]: +1 let's remove cosign and use rekor: the problem was reproducible with a SearchLogQuery

File: tests/sharding-e2e-test.sh [hunk 3]
```diff
@@ -251,6 +254,10 @@ ENTRY_ID_2=$($REKOR_CLI get --log-index 3 --rekor_server http://localhost:3000 -
 NUM_ELEMENTS=$(curl -f http://localhost:3000/api/v1/log/entries/retrieve -H "Content-Type: application/json" -H "Accept: application/json" -d "{ \"entryUUIDs\": [\"$ENTRY_ID_1\"]}" | jq '. | length')
 stringsMatch $NUM_ELEMENTS "1"
 
+# Make sure we can verify the blob we entered into the now-inactive shard
```
Comment (asraa) [inline tests/sharding-e2e-test.sh]: you can replace with a call to `./rekor verify --public-key blah --artifact blah --signature blah` with the artifact properties

**Review by asraa — APPROVED** [review]

**Pull request merged**